tests/backup/backup_helper.go

```diff
@@ -2076,6 +2149,176 @@ func restoreSuccessWithReplacePolicy(restoreName string, orgID string, retryDura
 	return err
 }
 
+func ValidateDataAfterRestore(expectedRestoredAppContexts []*scheduler.Context, restoreName string, appContext context.Context,
+	backupName string, namespaceMapping map[string]string, startTime time.Time) error {
+	var k8sCore = core.Instance()
+	var allBackupNamespaces []string
+	var dataBeforeBackup = make(map[string]map[string][][]string)
+	var dataAfterBackup = make(map[string]map[string][][]string)
+
+	var allRestoreHandlers []appDriver.ApplicationDriver
+	var allErrors []string
+
+	backupDriver := Inst().Backup
+	restoreInspectRequest := &api.RestoreInspectRequest{
+		Name:  restoreName,
+		OrgId: orgID,
+	}
+	restoreInspectResponse, err := backupDriver.InspectRestore(appContext, restoreInspectRequest)
+	if err != nil {
+		return err
+	}
+
+	backupUid, err := backupDriver.GetBackupUID(appContext, backupName, orgID)
+
+	backupInspectRequest := &api.BackupInspectRequest{
+		Name:  backupName,
+		Uid:   backupUid,
+		OrgId: orgID,
+	}
+	backupInspectResponse, _ := backupDriver.InspectBackup(appContext, backupInspectRequest)
```

Comment:
What will happen if we delete the backup before doing restore? We need to handle that here.